File: 1921 eliminateMaximum/1921.cpp
class Solution {
public:
    int eliminateMaximum(vector<int>& dist, vector<int>& speed) {
        int n=dist.size();
        vector<int>count(n,0);
        for(int i=0;i<n;i++){
            int Time=(dist[i]-1)/speed[i];
            if(Time<n)
                count[Time]++;
        }
        int cnt=0;
        for(int i=0;i<n;i++){
            cnt++;
            cnt-=count[i];
            if(cnt<0)
            return i+1;
        }
        return n;
    }
};

File: 1921 eliminateMaximum/1921.py
class Solution:
    def eliminateMaximum(self, dist: list[int], speed: list[int]) -> int:
        times=[(d-1)//s for d,s in zip(dist,speed)]
        times.sort()
        for i,t in enumerate(times):
            if t<i:
                return i
        return len(times)